internal/command/views/test.go

```diff
@@ -0,0 +1,99 @@
+package views
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/internal/command/arguments"
+	"github.com/hashicorp/terraform/internal/moduletest"
+)
+
+// Test renders outputs for test executions.
+type Test interface {
+	// Abstract should print an early summary of the tests that will be
+	// executed. This will be called before the tests have been executed so
+	// the status for everything within suite will be test.Pending.
+	//
+	// This should be used to state what is going to be tested.
+	Abstract(suite *moduletest.Suite)
+
+	// Conclusion should print out a summary of the tests including their
+	// completed status.
+	Conclusion(suite *moduletest.Suite)
+
+	// File prints out the summary for an entire test file.
+	File(file *moduletest.File)
+
+	// Run prints out the summary for a single test run block.
+	Run(run *moduletest.Run)
+}
+
+func NewTest(vt arguments.ViewType, view *View) Test {
+	switch vt {
+	case arguments.ViewJSON:
+		// TODO(liamcervante): Add support for JSON outputs.
+		panic("not supported yet")
+	case arguments.ViewHuman:
+		return &TestHuman{
+			view: view,
+		}
+	default:
+		panic(fmt.Sprintf("unknown view type %v", vt))
+	}
+}
+
+type TestHuman struct {
+	view *View
+}
+
+var _ Test = (*TestHuman)(nil)
+
+func (t *TestHuman) Abstract(_ *moduletest.Suite) {
+	// Do nothing, we don't print an abstract for the human view.
+}
+
+func (t *TestHuman) Conclusion(suite *moduletest.Suite) {
+	t.view.streams.Println()
+
+	counts := make(map[moduletest.Status]int)
+	for _, file := range suite.Files {
+		for _, run := range file.Runs {
+			count := counts[run.Status]
+			counts[run.Status] = count + 1
+		}
+	}
+
+	if suite.Status <= moduletest.Skip {
```

Comment:
It feels a little odd to use inequalities against state constants. I normally don't think of iota-derived constants as having an order. Did you consider using `suite.Status == moduletest.Pending || suite.Status == moduletest.Skip` and reject it?

Reply:
I can definitely update it to that here, but there's another function that also relies on the ordering. I added a comment there to highlight it.

Do you think it's worth me dropping any ordering assumptions to the status and updating the other function as well? I do think the iota implementation does enforce this, but I don't feel strongly about it.

Reply:
Nope, I hadn't spotted that and I think it's fine to rely on the ordering here. I'd maybe suggest adding commentary next to the type definition to clarify that order is significant and point to `Merge` as an example of that.